resource-rating-plugin.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Plugin Name: Resource Rating
+ * Description: A plugin to enable resource rating on posts and pages within WordPress
+ * Version: 0
+ * Author: Annabel Baynes
+ */
+
+//----------------------------------------------------------------------------------------------------------------------
+
+/**
+ * Description: Returns all of the resource ratings present on a post (identified by the post id sent through as a parameter).
+ *
+ * @param $post_id
+ *
+ * @return mixed
+ */
+function get_the_resource_ratings ($post_id) {
+	return get_post_meta($post_id, "resource_ratings");
+}
+
+/**
+ * Description: Adds a new rating to a post if it is a valid integer (post id and rating passed as parameters).
+ *
+ * @param $post_id
+ * @param $rating_value
+ */
+function submit_new_resource_rating ($post_id, $rating_value) {
+	if (filter_var($rating_value, FILTER_VALIDATE_INT)) {
+		add_post_meta( $post_id, "resource_ratings",$rating_value, false );
+	} else {
+		echo "Input is not an integer!";
+	}
+}
+
+/**
+ * Description: Deletes all resource ratings present on a post (identified by the post id sent through as a parameter).
+ *
+ * @param $post_id
+ */
+function delete_all_resource_ratings ($post_id) {
+	delete_post_meta($post_id, "resource_ratings");
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+
+/**
+ * Description: Updates the current ratings status on a post to be enabled (identified by the post id sent through as a parameter).
+ *
+ * @param $post_id
+ */
+function enable_resource_ratings ($post_id) {
+	update_post_meta($post_id, "_resource_rating_status", "ratings_enabled");
+}
+
+/**
+ * Description: Updates the current ratings status on a post to be disabled (identified by the post id sent through as a parameter).
+ *
+ * @param $post_id
+ */
+function disable_resource_ratings ($post_id) {
```

Comment:
`enable_resource_ratings` and `disable_resource_ratings` can be as a one function can be merged into one function.
Example, you can rename the function and call it, where ever needed.
```php
function toggle_resource_rating ($post_id, status){
update_post_meta($post_id, "_resource_rating_status", status);
}